tmspaymentsystem/src/paymentsystem/MerchantService.java

```diff
@@ -0,0 +1,164 @@
+package paymentsystem;
+
+import paymentsystem.exceptions.MerchantNotFoundException;
+import paymentsystem.exceptions.NoBankAccountsFoundException;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.nio.file.StandardOpenOption;
+import java.util.Comparator;
+import java.util.List;
+import java.util.stream.Stream;
+import java.nio.file.Path;
+
+import static java.nio.file.Files.createFile;
+
+public class MerchantService {
+    private List<Merchant> Merchants;
+    private static final String BANK_ACCOUNTS = "C:\\bank_account.txt";
+    private static final String MERCHANTS = "C:\\merchant.txt";
+    static Path bankAccountsFile;
+    static Path merchantsFile;
+
+    static {
+        try {
+            bankAccountsFile = createFile(Paths.get(BANK_ACCOUNTS));
+            merchantsFile = createFile(Paths.get(MERCHANTS));
+        } catch (IOException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+
+    public MerchantService(List<Merchant> merchants) {
+        this.Merchants = merchants;
+    }
+
+    public void setMerchants(List<Merchant> merchants) {
+        Merchants = merchants;
+    }
+
+    public void addBankAccount(Merchant merchant, BankAccount bankAccount) throws IllegalArgumentException {
+        boolean incorrectBankAccountNumber = Stream.of(bankAccount).anyMatch(c -> bankAccount.getAccountNumber().length() != 8 && bankAccount.getAccountNumber().matches("^[^0-9]*$"));
+        if (incorrectBankAccountNumber) {
+            throw new IllegalArgumentException("The length of account number should be eight digits without other characters!");
```

Comment:
throw new BankAccountNotAddedException("The length of account number should be eight digits without other characters!")